src/components/experimental/IndexProvider/IndexProvider.tsx

```diff
@@ -0,0 +1,205 @@
+import React, {useRef, useMemo, useCallback} from 'react';
+
+import {useI18n} from '../../../utilities/i18n';
+
+import {
+  IndexContext,
+  IndexRowContext,
+  IndexSelectionChangeContext,
+  SELECT_ALL_ITEMS,
+  IndexProviderProps,
+  SelectionType,
+  Range,
+} from './utilities/index';
+
+type HandleSelectionChange = (
```

Comment:
Should this type be used in your `IndexSelectionChangeContext` This one has a `sortOrder` and the other a `position`